dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/StatsTaskListGenerator.java

```diff
@@ -65,47 +65,41 @@ class StatsTaskListGenerator {
           "table-types-dtl",
           "used-space-details");
 
-  private static final ImmutableList<String> STATSPACK_NAMES =
-      ImmutableList.of("hist-cmd-types-statspack", "sql-stats-statspack");
-
   @Nonnull
   ImmutableList<Task<?>> createTasks(ConnectorArguments arguments, Duration queriedDuration) {
     ImmutableList.Builder<Task<?>> builder = ImmutableList.builder();
     builder.add(new DumpMetadataTask(arguments, scope.formatName()));
     builder.add(new FormatTask(scope.formatName()));
-    List<StatsJdbcTask> jdbcTasks = createJdbcTasks(queriedDuration);
-    builder.addAll(jdbcTasks);
 
-    QueryGroup awrAndCdb = QueryGroup.create(/* required= */ false, AWR, MULTI_TENANT);
-    QueryGroup awrAndNotCdb = QueryGroup.create(/* required= */ false, AWR, SINGLE_TENANT);
-    builder.add(ResultMessageTask.create(awrAndCdb, jdbcTasks));
-    builder.add(ResultMessageTask.create(awrAndNotCdb, jdbcTasks));
+    ArrayList<Task<?>> messageTasks = new ArrayList<>();
+    ImmutableList<OracleSqlListing> listsInOrder =
+        ImmutableList.of(awrCdb(), awrDba(), statspack());
+
+    for (OracleSqlListing sqls : listsInOrder) {
+      ImmutableList<Task<?>> messageTasksSoFar = copyOf(messageTasks);
+      ImmutableList<Task<?>> queryTasks =
```

Comment:
If I understand correctly, if only one AWR CDB task from the group fails, the ResultMessageTask will be skipped and all tasks (including succeeded) will be executed again with the DBA version. What will be the result in this case? We should not override successful results.

Reply:
This is true, Dumper assumes that any failure of the AWR CDB queries means that there is no multi-tenant setup - if any of the required native queries had failed, that would have been a different thing.
The assumption is due to Dumper's simplified architecture, where we don't specifically handle the errors, just try to obtain all information from the parts that do succeed.

In the case when:
- all required native queries succeeded
- there is CDB/PDBs support in the client DB
- some AWR CDB queries fail

It's unclear what Dumper should do. I'd assume that this is an invalid state and therefore we can't rely on the successful AWR CDBs results to be usable.